gnat2goto/driver/tree_walk.adb

```diff
@@ -5118,6 +5118,7 @@ package body Tree_Walk is
         (if Ekind (E) = E_Access_Subtype then Etype (E) else E);
       Type_Id : constant Symbol_Id := Intern (Type_Name);
    begin
+      Declare_Itype (E);
```

Comment:
This is the important bit; We now just proactively declare (possible) Itypes whenever we see them (`Declare_Itype` will take non-Itypes and just ignore them). This way, at least theoretically, we should always have types in the symbol table as soon as we reference them.